daemon/appinfo.c

```diff
@@ -902,6 +934,39 @@ appinfo_parse_desktop(appinfo_t *self)
             appinfo_set_application_name(self, tmp),
             g_free(tmp);
 
+        tmp = keyfile_get_string(ini, group, SAILJAIL_KEY_EXEC_DBUS, 0);
+        if( tmp ) {
+            /* As in libcontentaction, add invoker to the command line if it doesn't exist already. */
+            if( g_strstr_len(tmp, -1, "invoker") != tmp &&
+                    g_strstr_len(tmp, -1, "/usr/bin/invoker") != tmp) {
+                gchar *booster = keyfile_get_string(ini, DESKTOP_SECTION,
+                        NEMO_KEY_APPLICATION_TYPE, NULL);
+                if( booster == NULL || g_strcmp0(booster, "no-invoker") == 0 ) {
+                    /* Default booster type is "generic". This can be overridden via
+                      "X-Nemo-Application-Type=<boostertype>".
+                      "no-invoker" is synonymous to "generic" */
+                    g_free(booster);
+                    booster = g_strdup("generic");
+                }
+
+                gchar *single_instance = keyfile_get_string(ini, DESKTOP_SECTION,
+                        NEMO_KEY_SINGLE_INSTANCE, NULL);
+
+                gchar *exec = tmp;
```

Comment:
Done and then sort of undone because the context changed. I've moved the rewriting into a separate function because the set function was being called from within the if (tmp) which meant it wasn't being cleared if the entry was removed, and I looked at it and looked at it and could not see it.

Reply:
(I missed the re-use of tmp var on 1st read, but)  that separate function makes things much clearer.